core/trino-main/src/main/java/io/trino/operator/ScanFilterAndProjectOperator.java

```diff
@@ -356,8 +362,37 @@ public ProcessState<Page> process()
         }
     }
 
+    static class ProcessedBytesMonitor
+            implements Consumer<ProcessState<Page>>
+    {
+        private final SourcePage page;
+        private final LongConsumer processedBytesConsumer;
+        private long localProcessedBytes;
+
+        public ProcessedBytesMonitor(SourcePage page, LongConsumer processedBytesConsumer)
+        {
+            this.page = requireNonNull(page, "page is null");
+            this.processedBytesConsumer = requireNonNull(processedBytesConsumer, "processedBytesConsumer is null");
+            localProcessedBytes = page.getSizeInBytes();
+            processedBytesConsumer.accept(localProcessedBytes);
+        }
+
+        @Override
+        public void accept(ProcessState<Page> state)
+        {
+            update();
+        }
+
+        void update()
+        {
+            long newProcessedBytes = page.getSizeInBytes();
+            processedBytesConsumer.accept(newProcessedBytes - localProcessedBytes);
```

Comment:
when the new page is smaller than the one before the `update` the value of processedBytesConsumer will be decreased and even get to 0 when all rows were deleted. Does that make sense or we should use something like Math.abs(newProcessedBytes - localProcessedBytes)